src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal.OpenSsl
+{
+    internal sealed partial class PkcsPalOpenSsl : PkcsPal
+    {
+        public sealed override byte[] Encrypt(
+            CmsRecipientCollection recipients,
+            ContentInfo contentInfo,
+            AlgorithmIdentifier contentEncryptionAlgorithm,
+            X509Certificate2Collection originatorCerts,
+            CryptographicAttributeObjectCollection unprotectedAttributes)
+        {
+            using (SafeX509StackHandle originatorCertStack = Interop.Crypto.NewX509Stack())
+            using (SafeX509StackHandle issuerRecipientCerts = Interop.Crypto.NewX509Stack())
+            using (SafeX509StackHandle skidRecipientCerts = Interop.Crypto.NewX509Stack())
+            {
+                Interop.Crypto.CheckValidOpenSslHandle(issuerRecipientCerts);
+                Interop.Crypto.CheckValidOpenSslHandle(skidRecipientCerts);
+                Interop.Crypto.CheckValidOpenSslHandle(originatorCertStack);
+                                
+                ClassifyRecipients(recipients, issuerRecipientCerts, skidRecipientCerts);
+
+                foreach (X509Certificate2 cert in originatorCerts)
+                {
+                    using (SafeX509Handle certHandle = Interop.Crypto.X509Duplicate(cert.Handle))
+                    {
+                        if (!Interop.Crypto.PushX509StackField(originatorCertStack, certHandle))
```

Comment:
Since you're P/Invoking per cert here to push into the stack, but you're ultimately calling into an API that takes single certificates; do we need the X509Stack?  Or can we just loop through things in the presented order and classify them off to EnvelopedCmsAddIssuerSerialTransportRecipient(cms, cert.Handle) (and EnvelopedCmsAddSkidTransportRecipient(cms, cert.Handle))?

I guess the question is "what does the X509Stack buy us?", it feels like it's just buying complexity.